### PR TITLE
[Federation side branch] Use a feature flag to enable federated search

### PR DESCRIPTION
**Pull request by jlpadilla**
### Related Issue
https://issues.redhat.com/browse/ACM-0000

### Description of changes
- Use a feature flag `FEATURES_FEDERATED_SEARCH` to enable search federation.

### CODE REVIEW

File: pkg/federated/fedConfig.go [hunk 1]
```diff
@@ -21,42 +24,51 @@ type RemoteSearchService struct {
...
-	secret, err := client.CoreV1().Secrets("open-cluster-management").Get(context.TODO(), "search-global-token", metav1.GetOptions{})
+var cachedFedConfig = fedConfigCache{
+	lastUpdated: time.Time{},
+	fedConfig:   []RemoteSearchService{},
+}
 
-	if err != nil {
-		klog.Errorf("Error getting secret list: %s", err)
+func getFederationConfig(ctx context.Context, request *http.Request) []RemoteSearchService {
+	if cachedFedConfig.lastUpdated.IsZero() || cachedFedConfig.lastUpdated.Add(60*time.Second).Before(time.Now()) {
```
Comment (jlpadilla) [inline pkg/federated/fedConfig.go:38]: Add to config.

File: pkg/federated/federation-managed-hub-config.yaml [hunk 2]
```diff
@@ -38,6 +38,15 @@ spec:
           name: search-search-api
           weight: 100
 ---
+apiVersion: addon.open-cluster-management.io/v1alpha1
```
Comment (jlpadilla) [inline pkg/federated/federation-managed-hub-config.yaml:41]: Check if this will cause conflicts.

File: pkg/federated/setup.sh [hunk 3]
```diff
@@ -21,16 +22,43 @@ if [[ ! $REPLY =~ ^[Yy]$ ]]; then
...
+REQUIRED_RHACM_VERSION="2.10.0"
+oc get multiclusterhub -o yaml > /dev/null 2>&1
 if [ $? -ne 0 ]; then
-  echo "ERROR: MulticlusterGlobalHub is not installed in the target cluster. Exiting."
+  echo "❌ ERROR: Red Hat Advanced Cluster Management (MulticlusterHub) is not installed in the target cluster. Exiting."
   exit 1
+else
+  rhacm_version=$(oc get multiclusterhub -A -o yaml | yq '.items[0].status.currentVersion')
+  if ! printf "${REQUIRED_RHACM_VERSION}\n%s\n" $rhacm_version | sort -V -C; then
+    echo "❌ ERROR: Red Hat Advanced Cluster Management (MulticlusterHub) version ${rhacm_version} found, but ${REQUIRED_RHACM_VERSION} is required. Exiting."
```
Comment (jlpadilla) [inline pkg/federated/setup.sh:36]: Add "or later".

File: pkg/federated/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Federated Search
+
+Use federated search to query and combine results from multiple Red Hat Advanced Cluster Management.
+
+## Pre-requisites  
+1. Multicluster Global Hub Operator 1.0 or later.  
+2. Red Hat Advanced Cluster Management 2.10 or later.  
+
+## Setup
+Execute the script at `./setup.sh` to configure Global Search on the Global Hub cluster.
```
Comment (jlpadilla) [inline pkg/federated/README.md:10]: Add required RBAC permissions to execute this script.

File: pkg/federated/setup.sh [hunk 5]
```diff
@@ -21,16 +22,43 @@ if [[ ! $REPLY =~ ^[Yy]$ ]]; then
...
-# Validate that MulticlusterGlobalHub is installed.
-oc get multiclusterglobalhub > /dev/null 2>&1
+echo -e "Validating prerequisites...\n"
+
+# Validate that Red Hat Advanced Cluster Management (MulticlusterHub) 2.10.0 is installed.
+REQUIRED_RHACM_VERSION="2.10.0"
+oc get multiclusterhub -o yaml > /dev/null 2>&1
 if [ $? -ne 0 ]; then
-  echo "ERROR: MulticlusterGlobalHub is not installed in the target cluster. Exiting."
+  echo "❌ ERROR: Red Hat Advanced Cluster Management (MulticlusterHub) is not installed in the target cluster. Exiting."
```
Comment (jlpadilla) [inline pkg/federated/setup.sh:31]: Add "or user doesn't have permission."

**Review by SherinV — APPROVED** [review]
/lgtm

File: pkg/federated/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Federated Search
+
+Use federated search to query and combine results from multiple Red Hat Advanced Cluster Management.
+
+## Pre-requisites  
+1. Multicluster Global Hub Operator 1.0 or later.  
+2. Red Hat Advanced Cluster Management 2.10 or later.
```
[the remark this answers is not in the document]
Comment (jlpadilla) [reply pkg/federated/README.md:7]: resolved with commit https://github.com/stolostron/search-v2-api/pull/189/commits/c1c6f8aea9e70c6b531be736bd4f11a669222d98

File: pkg/federated/fedConfig.go (hunk 1 above)
Comment (jlpadilla) [reply pkg/federated/fedConfig.go:38]: resolved with commit https://github.com/stolostron/search-v2-api/pull/189/commits/c1c6f8aea9e70c6b531be736bd4f11a669222d98

File: pkg/federated/setup.sh (hunk 3 above)
Comment (jlpadilla) [reply pkg/federated/setup.sh:36]: resolved with commit https://github.com/stolostron/search-v2-api/pull/189/commits/c1c6f8aea9e70c6b531be736bd4f11a669222d98

File: pkg/federated/README.md (hunk 4 above)
Comment (jlpadilla) [reply pkg/federated/README.md:10]: resolved with commit https://github.com/stolostron/search-v2-api/pull/189/commits/c1c6f8aea9e70c6b531be736bd4f11a669222d98

**Pull request merged**